src/kernels/ipywidgets/localIPyWidgetScriptManager.node.ts

```diff
@@ -95,4 +67,59 @@ export class LocalIPyWidgetScriptManager extends BaseIPyWidgetScriptManager impl
     protected getWidgetScriptSource(source: Uri): Promise<string> {
         return this.fs.readLocalFile(source.fsPath);
     }
+    private async getNbExtensionsParentPathImpl(): Promise<Uri | undefined> {
+        const sourceNbExtensionsParentPath = this.nbExtensionsPathProvider.getNbExtensionsParentPath(this.kernel);
+        if (!sourceNbExtensionsParentPath) {
+            traceError(`Failed to get nbextensions parent path for ${this.kernel.kernelConnectionMetadata.id}`);
+            return;
+        }
+        const nbextensionsPath = path.join(sourceNbExtensionsParentPath.fsPath, 'nbextensions');
+        try {
+            const stopWatch = new StopWatch();
+            const kernelHash = getTelemetrySafeHashedString(this.kernel.kernelConnectionMetadata.id);
+            const baseUrl = Uri.joinPath(this.context.extensionUri, 'tmp', 'scripts', kernelHash, 'jupyter');
+            const filesPromise = this.fs.searchLocal('**/*.*', nbextensionsPath).catch((ex) => {
+                // Handle errors an ensure we don't crash the rest of the code just because we couldn't search for files.
+                traceError(`Failed to search for files in ${nbextensionsPath}`, ex);
+                return [];
+            });
+            // If we have previously copied the files, and the file count is the same, then don't overwrite them.
+            const previouslyCopiedFileCount = this.globalMemento.get<number>(this.getMementoKey(), 0);
+            if (previouslyCopiedFileCount > 0) {
+                const files = await filesPromise;
+                if (files.length === previouslyCopiedFileCount) {
```

Comment:
I guess the assumption is that files aren't updated? Or they're only updated by restarting the kernel? Was thinking you could upgrade a package and have a newer version of some file.

Reply:
yes, it just felt a little too slow to always copy the files.
in hindsight, this change might be risky, with the example you mentioned